test/cppsuite/test_harness/workload/database_operation.h

```diff
@@ -183,7 +183,13 @@ class database_operation {
                 /* Stop updating in case of an error. */
                 testutil_die(DEBUG_ERROR, "update_operation: cursor->next() failed: %d", ret);
             else {
-                testutil_check(cursors[i]->get_key(cursors[i], &key));
+                testutil_check(cursors[i]->get_key(cursors[i], &key_tmp));
+                /*
+                 * The retrieved key needs to be passed inside the update function. However, the
+                 * update API doesn't guarantee our buffer will still be valid once it
+                 * is called, as such we copy the buffer and then pass it into the API.
+                 */
+                key = key_value_t(key_tmp);
```

Comment:
Does key_value_t make a copy?

Reply:
It uses the copy constructor of the underlying type std::string with a const char* => std::string(const char*)